src/Symfony/Component/Messenger/EventListener/StopWorkerOnSignalListener.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ */
+class StopWorkerOnSignalListener implements EventSubscriberInterface
+{
+    private array $signals = [];
+    private ?LoggerInterface $logger;
+
+    public function __construct(int|array $signals, LoggerInterface $logger = null)
+    {
+        foreach ((array)$signals as $signal) {
+            $this->signals[$signal] = $this->strsignal($signal);
+        }
+
+        $this->logger = $logger;
+    }
+
+    private function strsignal(int $signal): string
+    {
+        foreach (get_defined_constants(true)['pcntl'] as $name => $num) {
+            // the _ is to ignore SIG_IGN and SIG_DFL and SIG_ERR and SIG_BLOCK and SIG_UNBLOCK and SIG_SETMARK, and maybe more, who knows
+            if ($num === $signal && 'SIG' === substr($name, 0, 3) && '_' !== $name[3]) {
+                return $name;
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf('The signal "%d" is not valid.', $signal));
+    }
```

Comment:
I would move this responsibility to the compilation/configuration step and just always have the signal int injected via the constructor.

Reply:
this is the case, but for logger, we need the name of the signal - so I need to reverse the logic - get the signal name by its id (int)